Merge latest RUC LSM into community develop and add new soil/veg data

tanyasmirnova opened the pull request:
## Description
This PR merges changes from RRFS_dev to UFS community develop related to:

1. Introducing new variables for fractional vegetation and soil data and and read this data in FV3GFS_io.F90.
2. Updated snow model in RUC LSM
3. Changes in the output related to LSM-produced variables.

(Instructions: this, and all subsequent sections of text should be removed and filled in as appropriate.)
Provide a detailed description of what this PR does.  
What bug does it fix, or what feature does it add?  
Is a change of answers expected from this PR?  



### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>



## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch


## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

tanyasmirnova committed:
Update .gitmodules and submodule pointers for ccpp/physics

tanyasmirnova committed:
Merged RRFS_dev changes needed for changes in the CCPP.

tanyasmirnova committed:
Update .gitmodules and submodule pointers for ccpp/physics for code review and testing

junwang-noaa commented on ccpp/driver/GFS_diagnostics.F90:
```diff
@@ -1823,12 +1824,11 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
-    ExtDiag(idx)%name = 'totsnwb_ave'
-    ExtDiag(idx)%desc = 'bucket surface snow precipitation rate'
-    ExtDiag(idx)%unit = 'kg/m**2/s'
+    ExtDiag(idx)%name = 'totsnwb'
```
We are using those fields for GFS post products for downstream users, if you need new total snow precipitation, please add a new variable, please do not replace the current fields.

tanyasmirnova replied:
@ericaligo-NOAA I only removed time-averaging of buckets, as it seems confusing. Are you sure you want to keep time-averaging and existing description for precipitation buckets?

tanyasmirnova replied:
@ericaligo-NOAA @junwang-noaa  Are we going to keep time-averaging of precipitation buckets? It does not make sense to me.

junwang-noaa replied:
It's interesting. But, yes, the global model is using these time-averaging of precipitation buckets and most operational centers do. So please don't remove/replace the existing fields. @WenMeng-NOAA

tanyasmirnova replied:
Done

junwang-noaa commented on ccpp/driver/GFS_diagnostics.F90:
```diff
@@ -1144,15 +1144,17 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
     ExtDiag(idx)%unit = '%'
     ExtDiag(idx)%mod_name = 'gfs_phys'
-    ExtDiag(idx)%time_avg = .TRUE.
+    !ExtDiag(idx)%time_avg = .TRUE.
```
The ExtDiag(idx)%time_avg and ExtDiag(idx)%cnvfac are used to get averaged snow cover (snowc_ave). the Instantaneous snowc is a different field, please do not replace the field as there is impact on downstream users. You can add a new variable "snowci" for instantaneous snow cover.

tanyasmirnova replied:
Ok, a new variable is added for instantaneous snow cover fraction.

yangfanglin replied:
Yes, Jun's comment is correct.  Changing it will have major impact on downstream models, forecasters and community users.

tanyasmirnova replied:
@yangfanglin @junwang-noaa Original time-averaged sow cover fraction is restored.

ericaligo-NOAA commented:
Hi Tanya, I believe that field already existed. The snow fields I added are tsnowp and tsnowpbEric
-------- Original message --------From: tanyasmirnova ***@***.***> Date: 3/9/23  12:40 PM  (GMT-05:00) To: NOAA-EMC/fv3atm ***@***.***> Cc: ericaligo-NOAA ***@***.***>, Mention ***@***.***> Subject: Re: [NOAA-EMC/fv3atm] Draft: Merge latest RUC LSM into community develop and add new soil/veg data (PR #633) 
@tanyasmirnova commented on this pull request.



In ccpp/driver/GFS_diagnostics.F90:
> @@ -1823,12 +1824,11 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
 
     idx = idx + 1
     ExtDiag(idx)%axes = 2
-    ExtDiag(idx)%name = 'totsnwb_ave'
-    ExtDiag(idx)%desc = 'bucket surface snow precipitation rate'
-    ExtDiag(idx)%unit = 'kg/m**2/s'
+    ExtDiag(idx)%name = 'totsnwb'

@ericaligo-NOAA I only removed time-averaging of buckets, as it seems confusing. Are you sure you want to keep time-averaging and existing description for precipitation buckets?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

ericaligo-NOAA commented:
I don't believe RRFS  has any use for it, but I'm not sure about global 
applications.

On 3/9/2023 1:29 PM, tanyasmirnova wrote:
>
> ***@***.**** commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In ccpp/driver/GFS_diagnostics.F90 
> <https://github.com/NOAA-EMC/fv3atm/pull/633#discussion_r1131437201>:
>
> > @@ -1823,12 +1824,11 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
>   
>       idx = idx + 1
>       ExtDiag(idx)%axes = 2
> -    ExtDiag(idx)%name = 'totsnwb_ave'
> -    ExtDiag(idx)%desc = 'bucket surface snow precipitation rate'
> -    ExtDiag(idx)%unit = 'kg/m**2/s'
> +    ExtDiag(idx)%name = 'totsnwb'
>
> @ericaligo-NOAA <https://github.com/ericaligo-NOAA> @junwang-noaa 
> <https://github.com/junwang-noaa> Are we going to keep time-averaging 
> of precipitation buckets? It does not make sense to me.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/NOAA-EMC/fv3atm/pull/633#discussion_r1131437201>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ALQ75MJORCQVNLNMM3XQQRDW3IOO3ANCNFSM6AAAAAAVUK5HMU>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>

tanyasmirnova committed:
Based on comments from reviewers, restored time-averaging of
precipitation buskets. Also, restored time-averaged snow cover fraction
and added instantaneous snow cover fraction.

SamuelTrahanNOAA committed:
bug fix from @tanyasmirnova for rrfs crashes during restart

tanyasmirnova committed:
Added allocation of rhonewsnow1 array.

tanyasmirnova committed:
Merged in the bug fix in writeing a RESTART that was crashing RRFS
retros.

tanyasmirnova committed:
Bug fix for NSST input

tanyasmirnova committed:
Merge latest changes to RUC LSM.

tanyasmirnova committed:
Changes in GFS_debug.F90 and RUC LSM to address reviewers' comments.

tanyasmirnova committed:
Moved internal RUC LSM options to the namelist options.

tanyasmirnova committed:
urface_snow_amount_vardens_over_land is changed to surface_snow_amount_assuming_variable_snow_density_over_land

tanyasmirnova committed:
point to top of ufsdev_ruclsm for ccpp/physics

SamuelTrahanNOAA committed:
incorrect character lengths in two calls

SamuelTrahanNOAA committed:
update from upstream develop

SamuelTrahanNOAA committed:
merge rrfs restart PR

joeolson42 committed:
update submodule pointer for ccpp-physics

SamuelTrahanNOAA committed:
changes from @tanyasmirnova to match baselines of non-ruc-lsm suites

joeolson42 committed:
update submodule pointer for ccpp-physics

tanyasmirnova committed:
Reverted changes to the default values of ivegsrc and isot.

SamuelTrahanNOAA committed:
merge upstream develop

SamuelTrahanNOAA committed:
Merge commit '4bab665' into ufsdev_ruclsm

SamuelTrahanNOAA committed:
merge joe/mynnsfc_24March

SamuelTrahanNOAA committed:
thread needs private copy of `num`

zach1221 commented:
Tests are complete on [#1646](https://github.com/ufs-community/ufs-weather-model/pull/1646), how is the review coming along on this PR? @junwang-noaa @SamuelTrahanNOAA

grantfirl commented:
https://github.com/ufs-community/ccpp-physics/pull/49 has been merged. @tanyasmirnova Please revert the changes to the  .gitmodules file, pull down the latest ufs/dev branch of https://github.com/ufs-community/ccpp-physics that contains the merged code and `git add ccpp/physics` within your fv3atm PR branch before `git commit` and `git push` to update the fv3atm PR branch. Once this is done, UFS code managers will review/approve the fv3atm PR, merge it, and then you'll repeat the procedure for ufs-weather-model (after pulling down the updated logs pushed by UFS code managers to your PR branch from GitHub).

SamuelTrahanNOAA committed:
point to ufs/dev ccpp-physics

SamuelTrahanNOAA committed:
point to ufs/dev ccpp-physics

tanyasmirnova commented:
@SamuelTrahanNOAA Sam, could you please revert the changes to .gitmodules file as Grant described?

SamuelTrahanNOAA commented:
This branch points to the ufs/dev branch of ccpp-physics and the .gitmodules has been reverted. You can commit now.

tanyasmirnova commented:
@SamuelTrahanNOAA Thank you, Sam.

grantfirl commented:
Thanks @SamuelTrahanNOAA. Sorry for bugging you about it @tanyasmirnova  -- I didn't realize Sam was handling this for you.

tanyasmirnova commented:
@grantfirl No problem, Grant.

jkbk2004 commented:
@DusanJovic-NOAA @BrianCurtis-NOAA @junwang-noaa please, go ahead to review and approve the pr.
